lib/Base/System/ClientState.vala

```diff
@@ -361,6 +362,11 @@ public partial class Backend.Client : Object {
     }
   }
 
+  public void register_session(Session session) throws Error {
+    sessions.add(session);
+    store_state ();
+  }
```

Comment:
This was replacing a plain `add()` elsewhere in the code. We could call `store_state` from outside, but I like the pattern of having a function to do it for us so that we don't forget if we have another location that adds sessions.

Reply:
Alright, that makes sense. But then, you should also add an `unregister` method doing the same when an session is removed.

Reply:
Yes, true. Although I can't remember seeing anything bothering to unregister sessions 😄